client/src/api/category.ts

```diff
@@ -0,0 +1,31 @@
+import axios from './axios';
+
+export default class Category {
+  static getAll = ({ limit, offset }: { limit: number; offset: number }) => {
+    return axios.post('/categories/all', { limit, offset });
+  };
+
+  static getByName = ({
+    name,
+    limit,
+    offset
+  }: {
+    name: string;
+    limit: number;
+    offset: number;
+  }) => {
+    return axios.post('/categories/search', { search: name, limit, offset });
```

Comment:
make it one endpoint that accepts search params

`/categories?name=sds`
`/categories`